lib/server.js

```diff
@@ -386,6 +387,91 @@ module.exports = function createServer(config, log) {
     }
   )
 
+  api.post('/checkCustomRule', function(req, res, next) {
+    const action = req.body.action
+    const ip = req.body.ip
+    const email = req.body.email
+
+    if (!ip || !action || !email) {
+      const err = {code: 'MissingParameters', message: 'action, ip email, are all required'}
+      log.error({op: 'request.checkCustomRule', action, ip, email})
+      res.send(400, err)
+      return next()
+    }
+
+    // Get all the custom rules that might apply to this action
+    const customRules = config.customRules
+    const checkRules = Object.keys(customRules).filter((key) => {
+      return customRules[key].actions.indexOf(action) > -1
+    })
```

Comment:
I suggest we pre-compute this at load time into a hash that maps action names to list of applicable rules, rather than doing a nested loop here on each request.